File: backend/Services/Parsers/KindaHardGolfParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class KindaHardGolfParser : BasicScoreResultParser
    {
        private ILogger<KindaHardGolfParser> _logger;

        public KindaHardGolfParser(ILogger<KindaHardGolfParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override string GameName => "Kinda Hard Golf";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($"kindahard.golf.+?📝 (?<{ScoreGroup}>\\d+)", RegexOptions.Singleline);
        protected override string? ExtraContent => null;
        public override string Url => "https://kindahard.golf";
    }
}

File: lambdas/src/parsers/kindahardgolf-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class KindaHardGolfParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Kinda Hard Golf';
  readonly helpText = null;
  protected readonly extraContent = null;
  readonly url = 'https://kindahard.golf';
  
  // Matches "kindahard.golf 01/25\n\n📝 10" - looks for 📝 emoji before score
  protected readonly parser = new RegExp(
    `kindahard\\.golf.+?📝 (?<${BasicScoreResultParser.SCORE_GROUP}>\\d+)`,
    's'
  );
}
